### PR TITLE
feat: add support for StreamedListObjects

### LINKED ISSUE #236: [Feature] Support for StreamedListObjects Endpoint
### Checklist

- [x] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

Support for the [StreamedListObjects](https://openfga.dev/api/service#/Relationship%20Queries/StreamedListObjects) endpoint needs to be added to the SDK.

### Describe the ideal solution

The SDK contains methods to call the StreamedListObjects API.

The implementation for this SDK must be aligned with the existing implementation of the Python SDK, with the exception of necessary language-specific conventions.

### Alternatives and current workarounds

The current workaround is to call the `/streamed-list-objects` API directly without using the SDK.

### References

- The current OpenFGA Python SDK does have support for StreamedListObjects as of [v0.9.1](https://github.com/openfga/python-sdk/releases/tag/v0.9.1) (PR https://github.com/openfga/python-sdk/pull/163).
- Parent issue: https://github.com/openfga/sdk-generator/issues/76.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by daniel-jonathan**
## Summary
Adds `streamedListObjects` method for retrieving unlimited objects via the streaming API. Node.js-only implementation with NDJSON parsing, error handling, and automatic resource cleanup.

Requires OpenFGA server v1.2.0+

Fixes #236

## Changes
- Add `streaming.ts` with NDJSON parser for Node.js streams
- Add `StreamedListObjectsResponse` interface to `apiModel.ts`
- Add `OpenFgaClient.streamedListObjects()` async generator method
- Add `createStreamingRequestFunction` to `common.ts` for streaming requests
- Export `parseNDJSONStream` from `index.ts`
- Add streaming tests and examples
- Update CHANGELOG with feature and version requirement

## Implementation
- Streams beyond 1000-object limit
- Proper error propagation through async iterators
- Automatic stream cleanup on early exit
- Supports Readable, AsyncIterable, string, Buffer, and Uint8Array inputs
- Telemetry integration via API layer

## Usage
```javascript
const fgaClient = new OpenFgaClient({ 
  apiUrl: "http://localhost:8080", 
  storeId: "01H0H015178Y2V4CX10C2KGHF4" 
});

for await (const response of fgaClient.streamedListObjects({
  user: "user:anne",
  relation: "owner",
  type: "document"
})) {
  console.log(response.object);
}
```

## Testing
All 153 tests passing (10/10 suites)
- 17 streaming tests covering parsing, errors, cleanup, edge cases
- 90.8% coverage on streaming.ts
- Live verified: 3-object and 2000-object streaming

## Related
- Fixes #236
- Parent issue: openfga/sdk-generator#76
- Related PR: openfga/sdk-generator#654

### CODE REVIEW

File: example/streamed-list-objects-local/streamedListObjectsLocal.mjs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { ClientConfiguration, OpenFgaClient, ConsistencyPreference } from "../../dist/index.js";
```
Comment (rhamzeh) [inline example/streamed-list-objects-local/streamedListObjectsLocal.mjs]: I don't think we need two examples for this, let's stick to this one only,

Also can you add the package.json for this example?

File: example/streamed-list-objects-local/streamedListObjectsLocal.mjs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                metadata: {
+                    relations: {
+                        can_read: {
+                            directly_related_user_types: [{ type: "user" }]
+                        }
+                    }
+                }
+            }
+        ]
+    };
```
Comment (rhamzeh) [inline example/streamed-list-objects-local/streamedListObjectsLocal.mjs]: Use the [syntax transformer](https://github.com/openfga/language/blob/main/pkg/js/README.md) here:

```suggestion
    const dslString = `
model
  schema 1.1

type user
type document
  relations
    can_read: [user]
`;
   const model = transformer.transformDSLToJSONObject(dslString);
```

File: example/streamed-list-objects/streamedListObjects.mjs [hunk 3]
```diff
@@ -30,24 +30,25 @@ async function main() {
...
-            { user: "user:anne", relation: "can_read", object: "document:2" },
-            { user: "user:anne", relation: "can_read", object: "document:3" }
-        ]
-    });
+    const writes = [];
+    for (let i = 1; i <= 2000; i++) {
+        writes.push({ user: "user:anne", relation: "can_read", object: `document:${i}` });
+    }
+    await fga.write({ writes });
+    console.log(`Wrote ${writes.length} tuples`);
```
[the remark this answers is not in the document]
Comment (daniel-jonathan) [reply example/streamed-list-objects/streamedListObjects.mjs]: In the Python SDK there is a similar approach to create a large list of tuples. I based it off that. That said, I do agree we can definitely create a better type of example. 

```python
async def write_tuples(openfga: OpenFgaClient, quantity: int) -> int:
    """
    Write a variable number of tuples to the temporary store.
    """

    chunks = quantity // 100

    for chunk in range(0, chunks):
        await openfga.write_tuples(
            [
                ClientTuple(
                    user="user:anne",
                    relation="owner",
                    object=f"document:{chunk * 100 + t}",
                )
                for t in range(0, 100)
            ],
        )

    return quantity
```

[the remark this answers is not in the document]
Comment (daniel-jonathan) [reply example/streamed-list-objects/streamedListObjects.mjs]: @aaguiarz 

## New Example Approach Summary

### The Change

**Old:** Write and query the same relation (direct tuples)
```
Write: user:anne can_read document:1-2000
Query: can_read
```

**New:** Write to base relations, query computed relation
```
Model:
  define owner: [user]
  define viewer: [user]
  define can_read: owner or viewer  ← COMPUTED

Write: 
  user:anne owner document:1-1000
  user:anne viewer document:1001-2000

Query: can_read (returns all 2000 via computation)
```

### Why

- **Shows OpenFGA's value**: Computed permissions from base relations
- **Demonstrates streaming need**: Computed relations return large result sets
- **Educational**: Real-world authorization pattern
